### PR TITLE
codeintel: Simplify PreciseIndex resolver

### PR DESCRIPTION
**Pull request by efritz**
Inline upload/index resolver into the combined resolver. We've removed all external uses of the previous node types.

## Test plan

Tested locally.

### CODE REVIEW

File: enterprise/internal/codeintel/shared/resolvers/precise_indexes.go [hunk 1]
```diff
@@ -61,19 +67,23 @@ func NewPreciseIndexResolver(
...
+		locationResolver: locationResolver,
+		traceErrs:        traceErrs,
+		upload:           upload,
+		index:            index,
 	}, nil
 }
 
+//
+//
+//
```
Comment (Strum355) [inline enterprise/internal/codeintel/shared/resolvers/precise_indexes.go:85]:
```
//
//
//
```

File: enterprise/internal/codeintel/shared/resolvers/precise_indexes.go [hunk 2]
```diff
@@ -86,52 +96,145 @@ func (r *preciseIndexResolver) ID() graphql.ID {
 	return relay.MarshalID("PreciseIndex", strings.Join(parts, ":"))
 }
 
-func (r *preciseIndexResolver) ProjectRoot(ctx context.Context) (resolverstubs.GitTreeEntryResolver, error) {
-	if r.uploadResolver != nil {
-		return r.uploadResolver.ProjectRoot(ctx)
+//
+//
+//
+//
```
Comment (Strum355) [inline enterprise/internal/codeintel/shared/resolvers/precise_indexes.go:102]:
```
//
//
//
//
```

File: enterprise/internal/codeintel/shared/resolvers/precise_indexes.go [hunk 3]
```diff
@@ -86,52 +96,145 @@ func (r *preciseIndexResolver) ID() graphql.ID {
...
 }
 
 func (r *preciseIndexResolver) Indexer() resolverstubs.CodeIntelIndexerResolver {
-	if r.uploadResolver != nil {
-		return r.uploadResolver.Indexer()
+	if r.index != nil {
+		// Note: check index as index fields may contain docker shas
+		return types.NewCodeIntelIndexerResolver(r.index.Indexer, r.index.Indexer)
+	} else if r.upload != nil {
+		types.NewCodeIntelIndexerResolver(r.upload.Indexer, "")
```
Comment (Strum355) [inline enterprise/internal/codeintel/shared/resolvers/precise_indexes.go:224]: :eye: smh smh

File: enterprise/internal/codeintel/shared/resolvers/precise_indexes.go (hunk 2 above)
Comment (efritz) [reply enterprise/internal/codeintel/shared/resolvers/precise_indexes.go:102]: Separating types of methods FOR YOUR CONVENIENCE

File: enterprise/internal/codeintel/shared/resolvers/precise_indexes.go (hunk 3 above)
Comment (efritz) [reply enterprise/internal/codeintel/shared/resolvers/precise_indexes.go:224]: oh jc

**Review by Strum355 — APPROVED** [review]

File: enterprise/internal/codeintel/shared/resolvers/precise_indexes.go (hunk 3 above)
Comment (efritz) [reply enterprise/internal/codeintel/shared/resolvers/precise_indexes.go:224]: Fixed.

**Pull request merged**